pydantic/fields.py

```diff
@@ -36,16 +51,117 @@ class Validator(NamedTuple):
 
 class Schema:
     """
-    Used to provide extra information about a field in a model schema.
+    Used to provide extra information about a field in a model schema. The parameters will be
+    converted to validations and will add annotations to the generated JSON Schema. Some arguments
+    apply only to number fields (``int``, ``float``, ``Decimal``) and some apply only to ``str``
+
+    :param default: since the Schema is replacing the field’s default, its first argument is used
+      to set the default, use ellipsis (``...``) to indicate the field is required
+    :param alias: the public name of the field
+    :param title: can be any string, used in the schema
+    :param description: can be any string, used in the schema
+    :param gt: only applies to numbers, requires the field to be "greater than". The schema
+      will have an ``exclusiveMinimum`` validation keyword
+    :param ge: only applies to numbers, requires the field to be "greater than or equal to". The
+    schema will have a ``minimum`` validation keyword
+    :param lt: only applies to numbers, requires the field to be "less than". The schema
+    will have an ``exclusiveMaximum`` validation keyword
+    :param le: only applies to numbers, requires the field to be "less than or equal to". The
+    schema will have a ``maximum`` validation keyword
+    :param min_length: only applies to strings, requires the field to have a minimum length. The
+    schema will have a ``maximum`` validation keyword
+    :param max_length: only applies to strings, requires the field to have a maximum length. The
+    schema will have a ``maxLength`` validation keyword
+    :param regex: only applies to strings, requires the field match agains a regular expression
+    pattern string. The schema will have a ``pattern`` validation keyword
+    :param **extra: any additional keyword arguments will be added as is to the schema
```

Comment:
TL;DR: Yes, it's still valid JSON Schema. Extensibility by adding extra keywords is part of the standard.

---

Additionally, there are some keywords that are part of the standard and are not declared as parameters of the `Schema` class. So, passing them as part of the `**extra` is the current only option to include them.

For example, the `examples` keyword could have a list of JSON "objects" (`dict`s) with examples of valid JSON "documents". ...I might create a new PR later with auto-generation of an example for each generated schema. But for now, they could be declared by hand by developers / Pydantic users using the `**extra`.

Also, I didn't add all the possible extra keywords as parameters to the `Schema` class because there are some keywords that would apply a validation that doesn't have an equivalent in Pydantic yet. For example, max and min size of an array (`list`). So, to not confuse users, I just added as parameters of `Schema` the things that actually implement a validation on data. And what I think are the most common extra informational (non-validation) fields: `title` and `description`.

Reply:
Ok